File: CORREDOR/Tipos Constructores/C#/MiProyecto/Estudiante.cs
class Estudiante : Persona, AccionesPersona{
    public int Edad;
    public string Curso;

    // Constructor con parámetros Copia
    public Estudiante(string nombre, string documento, int edad, string curso)
        : base(nombre, documento)
    {
        Edad = edad;
        Curso = curso;
    }
    // Constructor copia
    public Estudiante(Estudiante otro)
        : base(otro)   // llama al constructor copia de Persona
    {
        Edad = otro.Edad;
        Curso = otro.Curso;
    }
    public void Saludar(){
        Console.WriteLine("Hola, soy " + Nombre + ".");
    }
    public override void MostrarInformacion(){
        Console.WriteLine("Nombre: " + Nombre);
        Console.WriteLine("Documento: " + Documento);
        Console.WriteLine("Edad: " + Edad);
        Console.WriteLine("Curso: " + Curso);
    }
}

File: CORREDOR/Tipos Constructores/Java/Estudiante.java
package Java;

public class Estudiante extends Persona implements AccionesPersona{
    int edad;
    String curso;

    // Constructor con parámetros
    public Estudiante(String nombre, String documento, int edad, String curso) {
        super(nombre, documento);
        this.edad = edad;
        this.curso = curso;
    }

    // Constructor copia (recibe otro Estudiante)
    public Estudiante(Estudiante otro) {
        super(otro); // Llama al constructor copia de Persona
        this.edad = otro.edad;
        this.curso = otro.curso;
    }

    @Override
    public void saludar(){
        System.out.println("Hola soy " + nombre);
    }

    @Override
    void mostrarInformacion(){
        System.out.println("Nombre: " + nombre);
        System.out.println("Documento: " + documento);
        System.out.println("Edad: " + edad);
        System.out.println("Curso: " + curso);
    }
}